### PR TITLE
fix hugegraph test with apache release 1.2

### PR DESCRIPTION
**Pull request by liuxiaocs7**
1. `python deploy_start.py all` will deploy server and toolchain at the same time.
2. `pytest`

## Deploy
- [x] server  `python deploy_start.py server`
- [x] toolchain  `python deploy_start.py toolchain`

## graph function test

- [ ] aggregate (need cassandra backend)
- [x] hubble  (The compilation environment lacks node and yarn: https://github.com/apache/incubator-hugegraph-toolchain/pull/543)
- [x] loader  (version mismatch: https://github.com/apache/incubator-hugegraph-toolchain/pull/542)
- [ ] server
    - [ ] algorithm_olap  (not support in open source version)
    - [x] algorithm_oltp
    - [x] auth  (need start with auth)
    - [x] basic_operation
- [x] tools
- [x] ttl

## graph performance test

None

## CI

- [x] openSource and no auth (except hubble-load)
- [ ] openSource with auth

---

test data under `config/dataset.zip`
auth doc: https://hugegraph.apache.org/cn/docs/config/config-authentication/

### CODE REVIEW

**Comment by liuxiaocs7** [comment]
The version of hugegraph-client and hugegraph-server don't match

toolchain repo `revision`: https://github.com/apache/incubator-hugegraph-toolchain/blob/e8fd3e0a6c10d34b4bf5e92495f493c913d7895e/pom.xml#L100

image

File: src/config/basic_config.py [hunk 1]
```diff
@@ -4,13 +4,20 @@
...
 
 is_auth = False
 is_https = False
 
 # server
-server_git = {'branch': 'c0dff5b233e853716ca6f3f28a5cda05e6f3d639', 'url': 'https://github.com/hugegraph/hugegraph.git'}
+server_git = {
+    # 'branch': 'master',
+    'branch': '20d1e5228e39bd974079891fc713574fb14798c5',
+    'url': 'https://github.com/apache/incubator-hugegraph.git'
```
Comment (aroundabout) [inline src/config/basic_config.py:19]:
```suggestion
    'url': 'https://github.com/apache/hugegraph.git'
```
I think we can avoid name changed in future in the way

**Review by aroundabout — APPROVED** [review]

File: src/config/basic_config.py [hunk 2]
```diff
@@ -20,8 +27,14 @@
...
+test_password = {'tester': '123456'}
+
+# toolchain
+toolchain_git = {
+    'branch': 'b066b805630ca591809cfee2da2aae1395fc6b42',
+    'url': 'https://github.com/apache/incubator-hugegraph-toolchain.git'
+}
 
 # loader
 loader_git = {'branch': 'master', 'url': 'https://github.com/hugegraph/hugegraph-loader.git'}
```
[the remark this answers is not in the document]
Comment (liuxiaocs7) [reply src/config/basic_config.py]: Comments have been added and useless code removed

File: src/common/deploy_graph.py [hunk 3]
```diff
@@ -126,23 +140,41 @@ def __init__(self, obj):
...
+
+        # set properties && start hubble
+        hubble_package_dir_name = os.path.join(code_dir_path,
+                                               'apache-hugegraph-toolchain-incubating-1.0.0',
+                                               'apache-hugegraph-hubble-incubating-1.0.0')
+        # set_hubble_properties(hubble_package_dir_name, conf.graph_host, conf.hubble_port)
+        start_graph(hubble_package_dir_name, 'hubble')
+
     @staticmethod
     def hubble(self):
```
[the remark this answers is not in the document]
Comment (liuxiaocs7) [reply src/common/deploy_graph.py]: removed

**Review by aroundabout — APPROVED** [review]
LGTM

**Pull request merged**